app/composables/useAuth.ts

```diff
@@ -0,0 +1,101 @@
+import { ref } from "vue";
+import { useRouter } from "vue-router";
+
+const me = ref(null); // This variable stores the current logged in user data
+const error = ref(null);
+
+export function useAuth() {
+  const supabase = useSupabaseClient();
+  const router = useRouter();
+
+  async function signup(
+    email: string,
+    password: string,
+    username: string,
+    first_name: string,
+    last_name: string
+  ) {
+    error.value = null;
+    try {
+      const { data, error: signUpError } = await supabase.auth.signUp({
+        email,
+        password,
+      });
+
+      if (signUpError) throw signUpError;
+
+      const newUser = {
+        user_id: data.user?.id,
+        username,
+        first_name,
+        last_name,
+        email,
+      };
+
+      const response = await fetch("/api/users", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(newUser),
+      });
+
+      if (!response.ok) {
+        throw new Error("Failed to create user");
+      }
+
+      const insertedData = await response.json();
+
+      me.value = insertedData;
+      router.push("/about");
+    } catch (err: any) {
+      error.value = err.message;
+    }
+  }
+
+  async function login(email: string, password: string) {
+    error.value = null;
+    try {
+      const { data, error: loginError } =
+        await supabase.auth.signInWithPassword({
+          email,
+          password,
+        });
+
+      if (loginError) throw loginError;
+
+      console.log("User UUID:", data.user.id); // Supabase's user auth ID
+
+      const { data: userData, error: userError } = await supabase
+        .from("users")
+        .select("*")
+        .eq("user_id", data.user.id)
+        .single();
```

Comment:
Btw I noticed this here as well where we're accessing the user table directly instead of from the API